### DISCUSSION TITLE
I just broke the `rust-bitcoin` build

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#2273] [Announcements]
In the recent release of `rust-bitcoinconsensus` I managed to break the `rust-bitcoin` build. Note, cargo pulls the new version automatically (I guess because we use the semver compliant kludge `0.20.2-0.6.0` the last three numbers are ignored and the upgrade had the same first three?).

So all your PRs will break unless you patch the `bitcoin` toml to use `=`

```
bitcoinconsensus = { version = "=0.20.2-0.5.0", default-features = false, optional = true }
```

There is a PR up to fix it: https://github.com/rust-bitcoin/rust-bitcoinconsensus/pull/72

And yes, I totally did a release PR without testing it with `rust-bitcoin` :(

**Comment by dpc** [comment]
Would yanking help?

**Reply by tcharding** [reply]
Yes, I believe it would, good idea. @apoelstra can you handle that please when you wake up?

**Reply by tcharding** [reply]
Although, the release is not technically broken it just rugs `rust-bitcoin` - woops.

AFAICT this does not effect anyone unless they directly depend on `bitcoinconsensus` _and_ have an error type that embeds the `bitcoinconsensus::Error` - I'm guessing that is no-one. I just grepped the repos of Fedimint, Rust Lightning, and BDK and got no hits.

On further thought I personally wouldn't bother yanking, I'd just release the new point version.

**Reply by Kixunil** [reply]
If it rugs anything then it is by definition broken. But given the situation I think it's OK to wait with yanking until the fix is out.

**Reply by tnull** [reply]
@tcharding Unfortunately, `rust-lightning` does depend on `bitcoinconsensus` at least as a dev dependency (cf. https://github.com/lightningdevkit/rust-lightning/blob/0c677533fc3055163c1768ef2211fbf7317d65ab/lightning/Cargo.toml#L20). So until the release is yanked and/or the new one is out at least our tests and CI will be broken.

**Comment by sanket1729** [comment]
I believe Andrew and @BlueMatt can both yank the releases. Probably
@BlueMatt will be up earlier.

**Comment by Kixunil** [comment]
I wonder why you're recommending to pin using `Cargo.toml` rather than `Cargo.lock? I don't see any advantage and only disadvantages.

**Reply by tcharding** [reply]
Oh no reason, I just did the quickest thing to get a build working without thinking it through too much. I wasn't suggesting this as a permanent fix.

**Comment by apoelstra** [comment]
Morning all. Sure, I'll yank the latest bitcoinconsensus.

> I wonder why you're recommending to pin using `Cargo.toml` rather than `Cargo.lock? I don't see any advantage and only disadvantages.

`Cargo.toml` is automatic for downstream users. (But conversely, it can't be opted out of ... but in this case that's probably what we want since the upgrade was unintentional and broken.)

**Reply by apoelstra** [reply]
Yanked.
